### DISCUSSION TITLE
Order review points chronologically, oldest first

### DISCUSSION THREAD

**Discussion by drichardson** [tosdr/edit.tosdr.org#1006] [Ideas]
When I go to https://edit.tosdr.org/ I'm shown Points for you to review.

I noticed these appear to be sorted chonologically, newest first.

That means things that get backlogged may never get processed. It should be ordered by oldest first so that every point has a fair chance to be processed. Otherwise, it's like standing in a line where everyone gets to cut in front of you ;)

**Comment by Vinnl** [comment]
This is not an obvious choice; the reason we went for newest first is because old submissions are more likely to be outdated. Considering that this tool was introduced at a time when there was a big backlog of old issues, it seemed reasonable to first make sure we'd try to stay up-to-date with new points being added.

I don't have a strong opinion on it remaining that way, and I don't have a clear picture of how well we're currently managing to keep up with newly created points - just wanted to add some context to the status quo.

**Comment by drichardson** [comment]
Understood. So, the idea is that the bigger weakness is that the system does not identify out of date documents?

Ideally, if a document is out of date, the system should determine that automatically (e.g., via periodic re-crawl) and mark the documents pending (per your comment in #857).

Does the system ever re-crawl documents, or does it depend on human intervention to initiate a re-crawl? If you had that functionality, then anything in the moderator queue would not be outdated, at which point you should process oldest first for fairness sake.

P.S. Probably not new information for anyone, but just in case, the [Queuing Theory](https://en.wikipedia.org/wiki/Queueing_theory) entry on wikipedia has a good overview of the available options for processing line waiting tasks.

**Comment by drichardson** [comment]
Also, if there just isn't enough moderator/reviewer power to process the queue, then either:

- old, unreviewed points should be closed so that editors know they will never be processed (instead of leaving them in limbo indefinitely)
- old, unreviewed points should be automatically pinged to make sure a moderator sees them
- more moderators need to be added so that there is enough capacity to handle the review requests

I know some of these proposals require a bit of work both on systems and people processes, and I'm not sure how much interest there is in developing and maintaining tosdr. I only recently discovered tosdr because DuckDuckGo uses it in their privacy essentials, and I was a bit cynical that the only site that ever got an A was DuckDuckGo so I wanted to see how the project worked. I think the project is a good idea, but it needs a time investment in tools and community outreach (i.e. showing editors how to use the system properly, increasing the number of moderators, working on problems like this to optimize time issues spend in queues), if it's going to continue to mature and be a long lasting service. Sorry, I know this is a total meta point, and probably should be discussed elsewhere (but no one is responding to me on IRC ;)

I also think the project needs more transparency in showing exactly why a service gets a particular rating, since without it it seems like a DuckDuckGo promotional tool (or a Stack Overflow hating tool) ;)

**Comment by Vinnl** [comment]
> Understood. So, the idea is that the bigger weakness is that the system does not identify out of date documents?

Well, that's a weakness, but the particular problem here is that many of the points that were already in the database when the contribution tool was created, were not even linked to Documents, and hence still had to be linked in the first place. The plan was to comb through all of those and link them after all.

Perhaps the time has come that we should just decline all Pending points that are older than a certain point in time and have no Documents linked to them, and then implement your suggestions (oldest first, automatic recrawl).

But you've correctly identified the main pain point for ToS;dr: lack of time. The above needs developer time, on which we're a bit short at the moment. We'd also love to open a can of moderators, but they're hard to come by as well :)

The good news is that we do have [some funds available](https://opencollective.com/tosdr/), so we can sometimes fund people to come together and work on tosdr together, but unfortunately that by itself doesn't automatically get us new volunteers.
